extensions/amp-consent/0.1/amp-consent.js

```diff
@@ -723,6 +729,102 @@ export class AmpConsent extends AMP.BaseElement {
       opt_metadata['gdprApplies']
     );
   }
+
+  /**
+   * Checks if the payload from the `tcfapiCall` is valid.
+   * @param {JsonObject} payload
+   * @return {boolean}
+   */
+  isValidTcfApiCall_(payload) {
+    if (!isObject(payload)) {
+      user().error(TAG, `"tcfapiCall" is not an object: ${payload}`);
+      return false;
+    }
+    const {command, parameter, version} = payload;
+    if (!isEnumValue(TCF_POST_MESSAGE_API_COMMANDS, command)) {
+      user().error(
+        TAG,
+        `Unsupported command found in "tcfapiCall": ${command}`
+      );
+      return false;
+    }
+    if (parameter !== undefined) {
+      user().error(
+        TAG,
+        `Unsupported parameter found in "tcfapiCall": ${parameter}`
+      );
+      return false;
+    }
+    if (version != '2') {
+      user().error(TAG, `Found incorrect version in "tcfapiCall": ${version}`);
+      return false;
+    }
+    return true;
+  }
+
+  /**
+   * Maybe set up the __tfcApiLocator window and listeners.
+   *
+   * The window is a dummy iframe that signals to 3p iframes
+   * that the document supports the tcfPostMessage API.
+   */
+  maybeSetUpTcfPostMessageProxy_() {
+    if (!this.consentConfig_['exposesTcfApi']) {
+      return;
+    }
+    // Check if __tcfApiLocator API already exists (dirty AMP)
+    if (!this.win.frames[TCF_API_LOCATOR]) {
+      // Add window listener for 3p iframe PostMessages
+      this.win.addEventListener('message', this.boundHandleIframeMessages_);
+
+      // Set up the __tcfApiLocator window to singal PostMessage support
+      const iframe = this.element.ownerDocument.createElement('iframe');
+      iframe.setAttribute('name', TCF_API_LOCATOR);
+      toggle(iframe, false);
+      iframe.setAttribute('aria-hidden', true);
+      this.element.appendChild(dev().assertElement(iframe));
+    }
+  }
+
+  /**
+   * Listen to iframe messages and handle events.
+   *
+   * The listeners will listen for post messages from 3p
+   * iframes that have the following structure:
+   * {
+   *  "__tcfapiCall": {
+   *    "command": cmd,
+   *    "parameter": arg,
+   *    "version": v,
+   *    "callId": id,
+   *  }
+   * }
+   *
+   * @param {!Event} event
+   */
+  handleIframeMessages_(event) {
+    const data = getData(event);
+
+    if (
+      !data ||
+      !data['__tcfapiCall'] ||
+      !this.isValidTcfApiCall_(data['__tcfapiCall'])
+    ) {
+      return;
+    }
+
+    const payload = data.__tcfapiCall;
+    const cmd = payload.command;
+    // TODO (micajuineho): implement command methods
+    switch (cmd) {
```

Comment:
Before we actually implement the event handler, should we wrap the function in experiment? Or should we throw a user error at this point?

Reply:
Let's do an experiment.